analyses/chromosomal-instability/util/circos-plots.R

```diff
@@ -0,0 +1,338 @@
+        circlize::circos.genomicRect(region, value[[1]],
+          ytop = value[[1]] + 0.4,
+          ybottom = value[[1]] - 0.4,
+          col = value[[2]],
+          border = value[[2]]
+        )
+      }
+    )
+  }
+}
+
+circos_map_transloc <- function(df,
+                                add_track = FALSE,
+                                sample_names = NULL,
+                                samples_col = "samples",
+                                chr_col_1 = "chrom1",
+                                chr_col_2 = "chrom2",
+                                start_col_1 = "start1",
+                                start_col_2 = "start2",
+                                end_col_1 = "end1",
+                                end_col_2 = "end2") {
+  # Given two BED data.frames, plot translocations on a new or already exisiting circos plot.
+  #
+  # Args:
+  #   df: a data.frame with two sets of chromosomal coordinates indicating the translocations
+  #   add_track: If true, adds a track to a current plot in the global environment, if FALSE, starts a new plot.
+  #   sample_names: a character string that specifies values to keep from `samples_col` column.
+  #                 "all" keeps all samples in.
+  #   samples_col: a character string specifying the samples column which can be used to filter by.
+  #   chr_col_1/2: a character string that specifies the chromosomes column name for the from and to coordinates respectively.
+  #   start_col_1/2: a character string that specifies the start coordinate column name for the from and to coordinates respectively.
+  #   end_col_1/2: a character string that specifies the end coordinate column name for the from and to coordinates respectively.
+  #   y_val_1/2: The column name of the value you would like to plot for the from and to coordinates respectively.
+  #   palette: the color brewer palette you would like to be used. Run `RColorBrewer::display.brewer.all()` to see options.
+  #
+  # Returns:
+  #  Circos plot (or new circos plot track) of the data provided.
+
+  # Set up the df_1 how circlize wants it
+  bed_df_1 <- prep_bed(
+    df = df,
+    samples_col = samples_col,
+    sample_names = sample_names,
+    chr_col = chr_col_1,
+    start_col = start_col_1,
+    end_col = end_col_1
+  ) %>%
+    dplyr::mutate(value1 = rnorm(nrow(.)))
+
+  # Set up the df_2 how circlize wants it
+  bed_df_2 <- prep_bed(
+    df = df,
+    samples_col = samples_col,
+    sample_names = sample_names,
+    chr_col = chr_col_2,
+    start_col = start_col_2,
+    end_col = end_col_2
+  ) %>%
+    dplyr::mutate(value1 = rnorm(nrow(.)))
+
+  # If we aren't adding on a track to a current plot...
+  if (!add_track) {
+    # Initialize the plot
+    circlize::circos.clear()
+    circlize::circos.par(start.degree = 90)
+    circlize::circos.initializeWithIdeogram(species = "hg38")
+  }
+
+  circlize::circos.genomicLink(bed_df_1,
+    bed_df_2,
+    col = circlize::rand_color(nrow(bed_df_1),
```

Comment:
I don't like the random colors here. Allow a color column here too, perhaps?

Reply:
Yes, I can do that.